tools/local-environment/src/nomad.ts

```diff
@@ -672,216 +719,140 @@ export class Nomad {
     return;
   }
 
-  async deployAllNetworks(networks: Network[]): Promise<DeployArtifacts> {
+  async deployHubNSpoke(): Promise<void> {
     if (!this.host.isGovernor()) {
       this.host.setLocalGovernor(this.getDeployerKey(this.host)!.toAddress());
     }
 
-    const coreDeploysDeployed = await this.deployChains(networks);
-    const bridgeDeploysDeployed = await this.deployBridges(networks);
+    const [hubCoreDeployDeployed, spokesCoreDeploysDeployed] =
+      await this.deployHubNSpokeChains();
+    const [hubBridgeDeployDeployed, spokesBridgeDeploysDeployed] =
+      await this.deployHubNSpokeBridges();
 
-    const coreDeployArtifacts =
-      this.ejectCoreDeploysArtifacts(coreDeploysDeployed);
-    const bridgeDeployArtifacts = this.ejectBridgeDeploysArtifacts(
-      bridgeDeploysDeployed
+    this.updateArtifacts(
+      [hubCoreDeployDeployed, ...spokesCoreDeploysDeployed],
+      [hubBridgeDeployDeployed, ...spokesBridgeDeploysDeployed]
     );
 
-    const artifacts = this.mergeCoreAndBridgeArtifacts(
-      coreDeployArtifacts,
-      bridgeDeployArtifacts
+    this.setDeployed(
+      ...[hubCoreDeployDeployed, ...spokesCoreDeploysDeployed].map(
+        (n) => n.chain.domain
+      )
     );
 
-    this.setArtifacts(artifacts);
-
-    this.setDeployed(...networks.map((n) => n.domain));
-
-    return artifacts;
+    return;
   }
 
   async deployAdditionalNetworks(
-    host: Network,
-    oldNetworks: Network[],
-    newNetworks: Network[]
+    newNetworks: Network[],
+    connect?: Network[]
   ): Promise<void> {
     for (const newNetwork of newNetworks) {
-      await this.deployAdditionalNetwork(newNetwork, host, oldNetworks);
+      if (connect && connect.length) {
+        // await this.deployAdditionalNetworkCross(newNetwork, connect);
```

Comment:
remove `connect` param and this `if/else`

Reply:
https://github.com/nomad-xyz/nomad-monorepo/pull/17/commits/0a4f1c2357c07251712ed344633a1468f6efb311